source/client/client.cc

```diff
@@ -81,7 +81,12 @@ bool Main::run() {
     result = process->run(output_collector);
   }
   auto formatter = output_formatter_factory.create(options_->outputFormat());
-  std::cout << formatter->formatProto(output_collector.toProto());
+  absl::StatusOr<std::string> formatted_proto = formatter->formatProto(output_collector.toProto());
+  if (formatted_proto.ok()) {
+    std::cout << formatter->formatProto(output_collector.toProto());
+  } else {
+    ENVOY_LOG(error, "An error occured while formatting proto");
```

Comment:
should we force result to false here? if this code line get hit, that probably means we should indicate failure.

alternatively, we could consider propagating `absl::status_or` even further up (so we return that instead of the boolean we use return right now) if you'd be up for it, but feel free to not do this, I don't intent to bloat this PR.

Reply:
What shall be the status code for [1]?

Reply:
Consider setting result to false and fall trough, to have the `if(!result) {...` clause handle the logging/error handling. Minimizing the control logic helps keeping coverage up. If we want to specialise the message that gets logged we can declare/set that separately.